### PR TITLE
update the description of the annotated_arg/ptr spec

### PR DESCRIPTION
**Pull request by broxigarchen**
Address comments from Jessica and Joseph, updated description for some functions
Align the constructor for annotated_arg and annoated_ptr
Added a variadic template constructor

### CODE REVIEW

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_arg.asciidoc [hunk 1]
```diff
@@ -189,8 +189,16 @@ namespace sycl::ext::oneapi::experimental {
 template <typename T, typename PropertyListT = properties<>>
 class annotated_arg {
   public:
-    annotated_arg();
-    annotated_arg(const T& v_);
+    annotated_arg() noexcept;
+    annotated_arg(const T& v_, const properties<PropertyListT> &P = properties{}) noexcept;
+    template<typename T, typename... PropertyValueTs>
+    annotated_arg(T *Ptr, PropertyValueTs... props) noexcept;
```
Comment (tiwaria1) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_arg.asciidoc]:
```suggestion
    annotated_arg(const T& v, PropertyValueTs... props) noexcept;
```

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_arg.asciidoc [hunk 2]
```diff
@@ -237,11 +245,60 @@ Constructs an `annotated_arg` object which is default initialized.
...
+[source,c++]
+----
+template<typename T, typename... PropertyValueTs>
+annotated_arg(const T& v_, const PropertyValueTs... props) noexcept;
+----
+|
+Constructs an `annotated_arg` object from the input object `v_`. 
+
+The new property set `PropertyListT` contains all properties listed in `props`.
+If there were duplicate property in the list of `props`, the value of this property must be same.
```
Comment (tiwaria1) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_arg.asciidoc]:
```suggestion
If there are duplicate properties present in `props`, the values of the duplicate properties must be the same.
```
And the same wording correction in the descriptions below.

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc [hunk 3]
```diff
@@ -303,11 +309,24 @@ annotated_ptr(T *Ptr, const properties<PropertyListT> &P = properties{}) noexcep
...
+template<typename T, typename... PropertyValueTs>
+annotated_ptr(T *Ptr, const PropertyValueTs... props) noexcept;
+----
+|
+Constructs an `annotated_ptr` object. Does not allocate new storage. The
+underlying pointer is initialized with `Ptr`.
 
-The new property set `PropertyListT` must contain all properties from `P`,
-and if any common property takes a value, the value must be the same.
+The new property set `PropertyListT` contains all properties listed in `props`.
```
Comment (tiwaria1) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc]: Did spec reviewers request that this wording be changed?

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc [hunk 4]
```diff
@@ -303,11 +309,24 @@ annotated_ptr(T *Ptr, const properties<PropertyListT> &P = properties{}) noexcep
...
 |
 Constructs an `annotated_ptr` object. Does not allocate new storage. The
-underlying pointer is initialized with `Ptr`. `P` is used to specify the
-`PropertyListT` type on the class.
+underlying pointer is initialized with `Ptr`.
+
+The new property set `PropertyListT` contains all properties in `P`.
+If there were duplicate property in `P`, the value of this property must be same.
+
+// --- ROW BREAK ---
```
Comment (tiwaria1) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc:317]: I didn't check for this one but for both specs, please make sure that the order of the new ctors in this table is the same as the order in which they are listed in the class skeleton above. This was the case for the existing member functions.

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc [hunk 5]
```diff
@@ -210,6 +210,11 @@ using namespace sycl::ext::oneapi::experimental;
   // alignment of the pointer in bytes specified using the property 'alignment'
   auto arg_a = annotated_ptr(ptr, properties{alignment<4>});
 
+  // alignment in bytes and address bus width of the pointer specified using the
+  // property 'alignment' and 'awidth'.
```
Comment (tiwaria1) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc]:
```suggestion
  // properties 'alignment' and 'awidth'.
```

**Review by tiwaria1 — CHANGES_REQUESTED** [review]
Looks good, thanks Brox! Minor changes needed.

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc (hunk 3 above)
Comment (broxigarchen) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc]: `The underlying pointer is initialized with "Ptr". "P" is used to specify the "PropertyListT" type on the class. The new property set `PropertyListT` must contain all properties from "P"` This wording needs to be changed. Let me know if you like this new one or not

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc (hunk 4 above)
Comment (broxigarchen) [reply sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc:317]: done

**Review by tiwaria1 — APPROVED** [review]

**Pull request merged**